src/api-service/__app__/onefuzzlib/workers/scalesets.py

```diff
@@ -449,6 +449,29 @@ def _resize_shrink(self, to_remove: int) -> None:
         for node in nodes:
             node.send_stop_if_free()
 
+    def check_scaleset_size(self) -> None:
+        if self.state != ScalesetState.running:
+            return
+
+        size = get_vmss_size(self.scaleset_id)
+        if size is None:
+            logging.info(
+                SCALESET_LOG_PREFIX + "scaleset is unavailable. scaleset_id:%s",
+                self.scaleset_id,
+            )
+            self.set_shutdown(now=True)
```

Comment:
If the scaleset is unavailable, why does it make sense to `set_shutdown()`?

Reply:
The scaleset shutdown code handles cleaning up all of the resources for the scaleset.  This includes deleting nodes, stopping tasks on said nodes, deleting scaleset specific storage queues, etc.

Reply:
So is this just an optimization? If we don't call this here, what happens?

Reply:
The "Azure says this scaleset doesn't exist" error handling is to move to shutting it down.  I copied the comment from the other places we handle this same error condition.